pkg/kn/commands/service/service_create.go

```diff
@@ -131,17 +131,27 @@ func createService(client serving_v1alpha1_client.ServingV1alpha1Interface, serv
 }
 
 func replaceService(client serving_v1alpha1_client.ServingV1alpha1Interface, service *serving_v1alpha1_api.Service, namespace string, out io.Writer) error {
-	existingService, err := client.Services(namespace).Get(service.Name, v1.GetOptions{})
-	if err != nil {
-		return err
-	}
-	service.ResourceVersion = existingService.ResourceVersion
-	_, err = client.Services(namespace).Update(service)
-	if err != nil {
-		return err
+	var retries = 0
+	for true {
+		existingService, err := client.Services(namespace).Get(service.Name, v1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		service.ResourceVersion = existingService.ResourceVersion
```

Comment:
@rhuss : should we take these 3 operations off for loop? (get, err != nil and assigning ResourceVersion)
as we should get the ResourceVersion value same in each retry?

Reply:
No, it will be a new resource version so need to be set for each retry. Actually, the conflict occurs because of its a different resource version on the server than that we have sent. So we need to pick up this new resource version.

Reply:
ah okay! thanks!